subroutines.sh

```diff
@@ -3,39 +3,71 @@
 # subroutines.sh
 #
 
+
 # Function
 send_command() {
   vin=$1
-  shift
+  cmd=$2
+  cmd_name=$3
+
+  TESLA_CONTROL_CMD='/usr/bin/tesla-control -ble -vin $vin -key-name /share/tesla_blemqtt/${vin}_private.pem -key-file /share/tesla_ble_mqtt/${vin}_private.pem $cmd 2>&1'
 
+  # add a retry loop
   max_retries=5
-  for count in $(seq $max_retries); do
-    log_notice "Sending command $* to vin $vin, attempt $count/${max_retries}"
+  for tesla_ctrl_count in $(seq $max_retries); do
+
+    log_notice "Attempt $tesla_ctrl_count//${max_retries} sending $cmd_name to vin:$vin command:$cmd $2, attempt $tesla_ctrl_count/${max_retries}"
     set +e
     # shellcheck disable=SC2068
-    tesla_ctrl_out=$(tesla-control -vin $vin -ble -key-name /share/tesla_blemqtt/${vin}_private.pem -key-file /share/tesla_ble_mqtt/${vin}_private.pem $@ 2>&1)
+    tesla_ctrl_out=$(eval $TESLA_CONTROL_CMD)
     EXIT_STATUS=$?
     set -e
     if [ $EXIT_STATUS -eq 0 ]; then
-      log_info "tesla-control send command succeeded"
-      break
+      log_info "tesla-control command to vin:$vin was delivered"
+      return 0
     else
-      if [[ "$tesla_ctrl_out" == *"Failed to execute command: car could not execute command"* ]]; then
+      if [[ "$tesla_ctrl_out" == *"Failed to execute command to vin:$vin: car could not execute command"* ]]; then
```

Comment:
To avoid issues with spaces or quotes, maybe this is enough
`if [[ "$tesla_ctrl_out" == *"car could not execute command"* ]];`

Reply:
I just realized I added "vin:$vin" and that makes no sense !!!

Could we do something like this?!

```
failed_msg_regex="car could not execute command"
if echo $tesla_ctrl_out | grep -Eq $failed_msg_regex; then
  log_error "ERROR $failed_msg_regex"
else
  ...
fi
```

Reply:
If you tested it, then why not. Maybe more Unix elegant (though the initial version is more human readable :) )